src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py

```diff
@@ -862,65 +801,43 @@
     type: command
     short-summary: Get instance information about a VM.
     examples:
-        - name: Use a resource group and name to get instance view information of a VM.
-          text: az vm get-instance-view -g MyResourceGroup -n MyVm
 {0}
 """.format(vm_ids_example.format('Get instance views for all VMs in a resource group.', 'vm get-instance-view'))
 
 helps['vm list'] = """
     type: command
     short-summary: List details of Virtual Machines.
     long-summary: 'For more information on querying information about Virtual Machines, see https://docs.microsoft.com/en-us/cli/azure/query-az-cli2'
-    examples:
-        - name: List all VMs.
-          text: az vm list
-        - name: List all VMs by resource group.
-          text: az vm list -g MyResourceGroup
-        - name: List all VMs by resource group with details.
-          text: az vm list -g MyResourceGroup -d
```

Comment:
The third example is non-trivial *and* unusual as it deviates from the normal "list" pattern. I think it makes sense to call it out as we do in fact get questions about this.

Reply:
I'm not sure that these questions indicate a need for the example. The documentation for this command is quite poor:

```
    --show-details -d  : Show public ip address, FQDN, and power states. command will run slow.
```

This will mislead users to think that this is the _only_ information shown by this command when in fact this is _part_ of the data shown by the command. I bet that is where a substantial volume of these questions come from.

There is also no `long-summary` detail. The full summary from `-h` is so short that developers should see "If I want full details, add `-d`" without needing an example for this. My feeling on examples is that they represent more complex tasks or demonstrate mid- to high-level use of the CLI and should not be something that somebody can figure out from the general context provided by `-h`. In the case of `az vm list -g group -d` all of this information is summarized in 3 lines on the console.